### ISSUE TITLE
RequestInterceptor. error typedef missing described error parameter

### ISSUE THREAD

**Issue by basisbit** [Esri/jsapi-resources#201]
I am reporting an issue with 

- [x] TypeScript definitions
- [ ] another resource in this repository

### Problem
TypeScript Definitions for JSAPI 4.x
The RequestInterceptor.error type definition does not allow for an error object and thus makes it impossible to get the actual error message/object for logging. The comment of the error definition and also the jsapi 4 implementation actually describe and provide the desired error parameter detailing the reason of why the RequestInterceptor failed.

The faulty part of the definition of RequestInterceptor is this:
```
   /**
     * When an error occurs during the request processing, this function is called with an [Error](https://developers.arcgis.com/javascript/latest/api-reference/esri-core-Error.html) object giving the details about what happened. For example, this could be used to log specific errors occurring with layers or services.
     *
     * [Read more...](https://developers.arcgis.com/javascript/latest/api-reference/esri-config.html#RequestInterceptor)
     *
     *
     */
    error(): void;`
```
### Workaround
Our current workaround is to modify the type def locally, but that is additional work which has to be done every single time one updates the typedefs and new users of the lib might not know about this.

### Suggested Solution
After changing this part of the type definition to `error(error: __esri.Error): void;`, the error object can be used as described by the code description, solving the reported problem. The code implementation already provides this error parameter and thus it just works.

**Comment by NoashX** [comment]
Thank you for reporting this @basisbit, we'll take a look and get this fixed.

**Comment by NoashX** [comment]
We will have this fixed for the Spring release (4.15), and the update should be available on js-next sometime early next week. Thank you again for reporting this issue. I will now close it as resolved.

https://github.com/Esri/feedback-js-api-next/blob/master/CHANGELOG.md

**Comment by basisbit** [comment]
Thank you for the quick fix and fast response here! This was 100 times better than any support@esri.tld experience we had so far.
